### ISSUE TITLE
Visit remarks are randomly marked as confidential

### ISSUE THREAD

**Issue by KristinaBorch** [SORMAS-Foundation/SORMAS-Project#10607]
### Bug Description

One health department reported that serveral visit remarks are marked as confidential. The behaviour seems to appear randomly and cannot be narrowed down to specific user roles oder user role combinations. 
The bug cannot be reproduced from ourside and therefore we cannot provide any steps to reproduce. 

Some information from the health department: 
- teamleads have the user role Admin, Contact Supervisor and Surveillance Supervisor
- others have the user role Contact Supervisor, Surveillance Supervisor and Clinican 
- the visit remarks do not contain any special characteristics
- there are no similarities between cases and their visit remarks which are marked as confidential
- the responsible jurisdiction is in general the same for all cases
- no further changes on the user roles were reported

### Steps to Reproduce

1.
2.

### Screenshots

![image](https://user-images.githubusercontent.com/98821699/195094684-05e932ea-88ea-4215-8137-758c7aea4ffa.png)

### System Details

* Device:
* SORMAS version:
* Android version/Browser:
* Server URL:
* User Role:

**Comment by MateStrysewske** [comment]
@KristinaBorch Is it possible that the confidential visit was created by a different user than the other three? Have the other three visits been created by the same user who's taken the screenshot? Which specific user role combination does the user who's taken the screenshots have?

**Comment by KristinaBorch** [comment]
@MateStrysewske Yes, all visits were created by different users. But generel with the user role combinations Contact Supervisor, Surveillance Supervisor and Clinican or Admin, Contact Supervisor and Surveillance Supervisor. The user making the screenshot has the user roles clinican, contact supervisor and surveillance supervisor and did not create the other visitis.

**Comment by MateStrysewske** [comment]
Thanks!

**Comment by MateStrysewske** [comment]
Ah, I have a guess what might be the issue here, even though it's unlikely that this happens on a production system. Would like to verify that this is NOT the problem anyway: @KristinaBorch Can you check with the health department whether there's another case for the same person in a similar timeframe as this one, i.e. somewhere within 30 days of the 23th of February (likely within the 30 days after that date)? If so, the problem might be that this visit is primarily attached to that second case, and the user who's logged in doesn't have access to the personal/sensitive details of that case, and hence also not to those of the visit.
